kompendium-core/src/main/kotlin/io/bkbn/kompendium/Kontent.kt

```diff
@@ -80,42 +96,78 @@ object Kontent {
         clazz.isSubclassOf(Collection::class) -> handleCollectionType(type, clazz, cache)
         clazz.isSubclassOf(Enum::class) -> handleEnumType(clazz, cache)
         clazz.isSubclassOf(Map::class) -> handleMapType(type, clazz, cache)
-        else -> handleComplexType(clazz, cache)
+        else -> handleComplexType(type, clazz, cache)
       }
     }
   }
 
+  // TODO This shit is so gross, clean it up
   /**
    * In the event of an object type, this method will parse out individual fields to recursively aggregate object map.
    * @param clazz Class of the object to analyze
    * @param cache Existing schema map to append to
    */
-  private fun handleComplexType(clazz: KClass<*>, cache: SchemaMap): SchemaMap =
-    when (cache.containsKey(clazz.simpleName)) {
+  private fun handleComplexType(type: KType, clazz: KClass<*>, cache: SchemaMap): SchemaMap {
```

Comment:
This method needs to be broken up

Reply:
ehhh just commented instead this is a gross method and IDK if breaking it up will make it more clear